### PR TITLE
Add DeleteQueryIndexOperation to delete indexes

### LINKED ISSUE #9: Support deleting query indexes
Support deleting Cloudant Query indexes, this should be based on the interface in [objective-cloudant](https://github.com/cloudant/objective-cloudant/blob/master/ObjectiveCloudant/Operations/CDTDeleteQueryIndexOperation.h).

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Add the ability to delete query indexes
## How
- Created new DeleteQueryIndexOperation
- Created IndexType enum to provide type safety around index types.

## Testing
- Tests added to DeleteQueryIndexOperation

## Reviewers
reviewer: @brynh

## Issues

Fixes #9

### CODE REVIEW

File: Source/DeleteQueryIndexOperation.swift [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+	- Requires: The `designDoc`, `indexName` and `type` properties to be set.
+	
+	Example usage:
+	```
+	let deleteIndex = DeleteQueryIndexOperation()
+	deleteIndex.designDoc = "exampleDesignDoc"
+	deleteIndex.indexName = "exampleIndexName"
+	deleteIndex.type = .JSON
+	deleteIndex.completionHanlder = {(response, httpInfo, error) in
```
Comment (brynh) [inline Source/DeleteQueryIndexOperation.swift]: Typo `completionHanlder` -> `completionHandler`

File: Tests/DeleteQueryIndexTests.swift [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        db = client![dbName!]
+    }
+
+    override public func tearDown() {
+        super.tearDown()
+        OHHTTPStubs.removeAllStubs()
+    }
+
+    func testCanDeleteJSONIndex() {
+    	let deleteExpecation = self.expectation(withDescription: "delete json index")
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: Typo `deleteExpecation` -> `deleteExpectation`

File: Tests/DeleteQueryIndexTests.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    		}
+    		XCTAssertNil(error)
+    		deleteExpecation.fulfill()
+    	}
+    	self.db?.add(operation: deleteIndex)
+    	self.waitForExpectations(withTimeout:10.0, handler:nil)
+    }
+
+    public func testCanDeleteTextIndex() {
+    	let deleteExpecation = self.expectation(withDescription: "delete json index")
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: Typo `deleteExpecation` -> `deleteExpectation`

File: Tests/DeleteQueryIndexTests.swift [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    	let deleteIndex = DeleteQueryIndexOperation()
+    	deleteIndex.designDoc = "ddoc"
+    	deleteIndex.indexName = "jsonIndex"
+    	deleteIndex.type = .JSON
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: This should be `.Text` not `.JSON`.

File: Tests/DeleteQueryIndexTests.swift [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    		deleteExpecation.fulfill()
+    	}
+    	self.db?.add(operation: deleteIndex)
+    	self.waitForExpectations(withTimeout:10.0, handler:nil)
+    }
+
+    public func testValidationMissingDdoc() {
+		let deleteIndex = DeleteQueryIndexOperation()
+    	deleteIndex.indexName = "jsonIndex"
+    	deleteIndex.type = .Text
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: It doesn't really matter, but it looks a bit odd to have an `indexName` of `jsonIndex` which is of `type` of `.Text`.

Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: done in 6760fab

File: Tests/DeleteQueryIndexTests.swift (hunk 4 above)
Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: done in 6760fab

File: Tests/DeleteQueryIndexTests.swift (hunk 3 above)
Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: done in 6760fab

File: Tests/DeleteQueryIndexTests.swift (hunk 2 above)
Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: done in 6760fab

File: Source/DeleteQueryIndexOperation.swift (hunk 1 above)
Comment (rhyshort) [inline Source/DeleteQueryIndexOperation.swift]: done in 6760fab

Comment (brynh) [inline Source/DeleteQueryIndexOperation.swift]: This wasn't done in 6760fab.

Comment (rhyshort) [inline Source/DeleteQueryIndexOperation.swift]: oops, really done in df42d6a this time

File: Tests/DeleteQueryIndexTests.swift [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+//  and limitations under the License.
+//
+
+import Foundation
+import XCTest
+import OHHTTPStubs
+@testable import SwiftCloudant
+
+public class DeleteQueryIndexTests: XCTestCase {
+	var client: CouchDBClient? = nil;
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: Inconsistency - semicolons are not used to terminate lines elsewhere.

File: Tests/DeleteQueryIndexTests.swift [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    var dbName: String? = nil
+    var db: Database?
```
Comment (brynh) [inline Tests/DeleteQueryIndexTests.swift]: Inconsistency - this isn't explicitly initialised to nil (though my understanding is that's the default for an optional in Swift).

File: Tests/DeleteQueryIndexTests.swift (hunk 6 above)
Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: fixed in ad04334

File: Tests/DeleteQueryIndexTests.swift (hunk 7 above)
Comment (rhyshort) [inline Tests/DeleteQueryIndexTests.swift]: fixed in ad04334

**Comment by brynh** [comment]
+1

**Pull request merged**